Zend/zend_API.h

```diff
@@ -682,6 +685,50 @@ END_EXTERN_C()
 #define ZEND_GINIT_FUNCTION			ZEND_MODULE_GLOBALS_CTOR_D
 #define ZEND_GSHUTDOWN_FUNCTION		ZEND_MODULE_GLOBALS_DTOR_D
 
+static zend_always_inline int zend_try_assign_ex(zval *zv, zval *arg, zend_bool silent, zend_bool strict) {
+	zend_type ref_type;
+	if (EXPECTED(Z_ISREF_P(zv)) && UNEXPECTED(ZEND_TYPE_IS_SET(ref_type = Z_REFTYPE_P(zv)))) {
+		zval tmp;
+		ZVAL_COPY_VALUE(&tmp, arg);
+		if (zend_verify_ref_type_assignable_zval(ref_type, &tmp, strict)) {
+			zv = Z_REFVAL_P(zv);
+			zval_ptr_dtor(zv);
+			ZVAL_COPY_VALUE(zv, &tmp);
+		} else {
+			if (!silent) {
+				zend_throw_ref_type_error(ref_type, &tmp);
+			}
+			zval_ptr_dtor(&tmp);
+			return FAILURE;
+		}
+	} else {
+		ZVAL_DEREF(zv);
+		zval_ptr_dtor(zv);
+		ZVAL_COPY_VALUE(zv, arg);
+	}
+
+	return SUCCESS;
+} 
+
+static zend_always_inline int zend_try_assign(zval *zv, zval *arg) {
+	return zend_try_assign_ex(zv, arg, 0, ZEND_ARG_USES_STRICT_TYPES());
+}
+
+#define ZEND_TRY_ASSIGN(func, zv, ...) \
+	do { \
+		zval _zv; \
+		func(&_zv, ##__VA_ARGS__); \
+		zend_try_assign(zv, &_zv); \
+        } while (0) 
+
```

Comment:
This piece is not portable, so the compilation fails at least on windows. I've just reworked another similar place and generalized the mechanism for macros containing VA. If you please pull current master and apply https://gist.github.com/weltling/c7926ae30b6e39edbe9f5174a76c2061, then the VC compilation should work. At least it fixed it for the local build. Doing further testing currently.

Thanks.

Reply:
I've fixed this in a different way in https://github.com/php/php-src/pull/3313/commits/2d47e84d4b10b838a26a7d38bb4112ae17fe60c5. I think this black magic is not worth it to save a couple lines of code :)

Reply:
The simplest is the best, as usual :) Thanks!